compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -992,11 +980,42 @@ private STNode parseReturnTypeDescRhs(STNode metadata, STNode functionKeyword, S
         funcSignature = validateAndGetFuncParams((STFunctionSignatureNode) funcSignature);
 
         STNode funcDef =
-                createFuncDefOrMethodDecl(metadata, functionKeyword, isObjectMethod, name, funcSignature, qualifiers);
+                createFuncDefOrMethodDecl(metadata, functionKeyword, isObjectMember, name, funcSignature, qualifiers);
         endContext();
         return funcDef;
     }
 
+    private STNode parseVarDeclWithFunctionType(STNode typeDesc, boolean isObjectMember, STNode qualifiers,
+                                                STNode metadata) {
+        STNodeList qualifierList = (STNodeList) qualifiers;
+        STNode visibilityQualifier = STNodeFactory.createEmptyNode();
+        // qualifiers are not allowed except for object filed
+        for (int position = 0; position < qualifierList.size(); position++) {
+            STNode qualifier = qualifierList.get(position);
+            if (isObjectMember) {
+                // public or private qualifier allowed in object field.
+                if (qualifier.kind == SyntaxKind.PUBLIC_KEYWORD || qualifier.kind == SyntaxKind.PRIVATE_KEYWORD) {
+                    visibilityQualifier = qualifier;
+                } else {
+                    // if remote or transactional qualifier present
+                    typeDesc = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(typeDesc, qualifier,
+                            DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED);
```

Comment:
Related above comment. 
```suggestion
                    typeDesc = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(typeDesc, qualifier,
                            DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED, ((STToken) qualifier).text()));
```